cloudferrylib/utils/files.py

```diff
@@ -89,4 +90,19 @@ def _unlink_dir(dirname):
         return "rm -rf {dir}".format(dir=dirname)
     else:
         raise RuntimeError('Wrong dirname %s, stopping' % dirname)
-    return
+
+
+class RemoteTempDir(object):
+    """Creates remote temp dir using `mktemp` and removes it on scope exit"""
+
+    def __init__(self, runner):
+        self.runner = runner
+
+    def __enter__(self):
+        create_temp_dir = 'mktemp -d'
```

Comment:
We already have GetTempDir to get directory name, it can be used here. Or use the same format of directory "mktemp -dt %s_XXXX" % self.prefix